packages/components/menu/src/menu-item.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <li
+    class="el-menu-item"
+    role="menuitem"
+    tabindex="-1"
+    :style="paddingStyle"
+    :class="{
+      'is-active': active,
+      'is-disabled': disabled,
+    }"
+    @click="handleClick"
+  >
+    <el-tooltip
+      v-if="
+        parentMenu.type.name === 'ElMenu' &&
+        rootMenu.props.collapse &&
+        $slots.title
+      "
+      :effect="Effect.DARK"
+      placement="right"
+    >
+      <template #content>
+        <slot name="title" />
+      </template>
+      <div
+        :style="{
+          position: 'absolute',
+          left: 0,
+          top: 0,
+          height: '100%',
+          width: '100%',
+          display: 'inline-block',
+          boxSizing: 'border-box',
+          padding: '0 20px',
+        }"
+      >
+        <slot />
+      </div>
+    </el-tooltip>
+    <template v-else>
+      <slot />
+      <slot name="title" />
+    </template>
+  </li>
+</template>
+
+<script lang="ts">
+import {
+  defineComponent,
+  computed,
+  onMounted,
+  onBeforeUnmount,
+  inject,
+  getCurrentInstance,
+  toRef,
+  reactive,
+} from 'vue'
+import ElTooltip from '@element-plus/components/tooltip'
+import { Effect } from '@element-plus/components/popper'
+import { throwError } from '@element-plus/utils/error'
+import useMenu from './use-menu'
+import { menuItemEmits, menuItemProps } from './menu-item'
+
+import type { MenuItemRegistered, MenuProvider, SubMenuProvider } from './types'
+
+const COMPONENT_NAME = 'ElMenuItem'
+export default defineComponent({
+  name: COMPONENT_NAME,
+  components: {
+    ElTooltip,
+  },
+
+  props: menuItemProps,
+  emits: menuItemEmits,
+
+  setup(props, { emit }) {
+    const instance = getCurrentInstance()!
+    const rootMenu = inject<MenuProvider>('rootMenu')
+    if (!rootMenu) throwError(COMPONENT_NAME, 'can not inject root menu')
```

Comment:
**Non-blocking:**  Maybe we can make `rootMenu` a `symbol` as we did for `elFormKey`, and extract them into `@element-plus/tokens`

This can be changed in a followup PR. ;)

Reply:
Yes, I tried, but `@element-plus/token` build script seems broken. So I need fix build script first.

Reply:
Then how did the forms and buttons working 😕

Reply:
Only God knows. 😕

Reply:
TypeScript emit errors.